lib/src/settings.rs

```diff
@@ -42,6 +42,7 @@ pub struct RepoSettings {
 pub struct GitSettings {
     pub auto_local_branch: bool,
     pub abandon_unreachable_commits: bool,
+    pub private_commits: Option<String>,
```

Comment:
nit: it's unclear whether this config should be in library `GitSettings`, but I think it is closer to `git.push`/`git.fetch`, so belongs to the CLI.

Reply:
Oh, I didn't notice this was in the lib. I just saw where the other two `git.` settings were and put it alongside. I'll try to move this out to the cli.